org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/model/JDIThread.java

```diff
@@ -1478,9 +1484,21 @@ private boolean handleSuspendForBreakpointInternal(JavaBreakpoint breakpoint) {
 			}
 		}
 
+		try {
+			if (breakpoint.getSuspendPolicy() == IJavaBreakpoint.RESUME_ON_SUSPEND && resumeCondition) {
```

Comment:
I would put `resumeCondition` first, just in case the first one would throw errors, if the first one is false, it will not spam the log.